freqtrade/rpc/rpc.py

```diff
@@ -717,7 +717,8 @@ def _exec_forcesell(trade: Trade) -> None:
             return {'result': f'Created sell order for trade {trade_id}.'}
 
     def _rpc_forcebuy(self, pair: str, price: Optional[float], order_type: Optional[str] = None,
-                      stake_amount: Optional[float] = None) -> Optional[Trade]:
+                      stake_amount: Optional[float] = None,
+                      buy_tag: Optional[str] = None) -> Optional[Trade]:
```

Comment:
while `buy_tag` is generally a user-controlled field (it comes from the strategy) - i fear this has potential negative side-effects.

In theory, you could insert a malicious string here, which might (!) be shown in a ui component - effectively resulting in a "stored XSS" attack.

That's not possible with other fields i'm aware of - so it's something i'll need to think about a bit more.

Reply:
good point, I can add some method that will strip out malicious tags from this string maybe? Let me know if you have any other idea

Reply:
that'll be pretty difficult - there's too much possibility to obfuscate tags or whatnot.
it'll also depend on the usecase - in freqUI, vue will render this as string and you have to jump through hoops to have it render a "real" injected script-tag ...

but freqUI is potentially not the only API consumer - and it'll be difficult to protect against unknown consuming languages.